pkg/shell-operator/operator.go

```diff
@@ -631,8 +650,12 @@ func (op *ShellOperator) handleRunHook(t task.Task, taskHook *hook.Hook, hookMet
 		return err
 	}
 
-	if result != nil && result.Usage != nil {
-		taskLogEntry.Debugf("Usage: %+v", result.Usage)
+	if result == nil {
+		return nil
+	}
+
+	if result.Usage != nil {
+		taskLogEntry.Debug("Usage", slog.String("value", fmt.Sprintf("%+v", result.Usage)))
```

Comment:
The same story, result is nil only if some specific error takes place. If we return nil for such a result, it marks the run as successful, which is barely true.
```
} else {
                        success = 1.0
                        taskLogEntry.Infof("Hook executed successfully")
                        res.Status = "Success"
                }
```

Reply:
I just removed this check, at least this is not within the scope of the current task.